test/js/specs/unit/models/maps/GeoPoint.spec.js

```diff
@@ -54,8 +82,107 @@ define([
         });
         point.isValid().should.be.false;
       });
+
+      it("includes a height-specific validation error", function () {
+        var point = new GeoPoint({
+          latitude: 0,
+          longitude: 0,
+          height: "foo"
+        });
+
+        point.isValid();
+
+        expect(point.validationError).to.deep.equal({
+          height: 'Invalid height. Must be a number.'
+        });
+      });
+    });
+
+    describe('Instantiation', () => {
+      describe('from a good string', () => {
+        it('uses the user\'s search query when zooming', () => {
+          const geoPoint = GeoPoint.fromString('13,37');
+
+          expect(geoPoint.attributes.latitude).to.equal(13);
+          expect(geoPoint.attributes.longitude).to.equal(37);
+        });
+
+        it('accepts two space-separated numbers', () => {
+          const geoPoint = GeoPoint.fromString('13 37');
+
+          expect(geoPoint.attributes.latitude).to.equal(13);
+          expect(geoPoint.attributes.longitude).to.equal(37);
+        });
+
+        it('accepts input with \'-\' signs', () => {
+          const geoPoint = GeoPoint.fromString('13,-37');
+
+          expect(geoPoint.attributes.latitude).to.equal(13);
+          expect(geoPoint.attributes.longitude).to.equal(-37);
+        });
+
+        it('accepts input of with \'+\' signs', () => {
+          const geoPoint = GeoPoint.fromString('13,+37');
+
+          expect(geoPoint.attributes.latitude).to.equal(13);
+          expect(geoPoint.attributes.longitude).to.equal(37);
+        });
+
+        it('accepts input with a trailing comma', () => {
+          const geoPoint = GeoPoint.fromString('13,37,');
+
+          expect(geoPoint.attributes.latitude).to.equal(13);
+          expect(geoPoint.attributes.longitude).to.equal(37);
+        });
+      });
+
+      describe('from a bad string', () => {
+        it('shows an error when only a single number is entered', () => {
+          expect(() => {
+            GeoPoint.fromString('13');
+          }).to.throw(Error, /Try entering a search query with two numerical values/);
+        });
+
+        it('shows an error when non-numeric characters are entered', () => {
+          expect(() => {
+            GeoPoint.fromString('13,37a');
+          }).to.throw(Error, /Try entering a search query with two numerical values/);
+        });
+      });
     });
 
+    describe("Detecting latitude, longitude in string", () => {
+      it('accepts empty string', () => {
+        expect(GeoPoint.couldBeLatLong('')).to.be.true;
```

Comment:
I just mean for this function to disqualify strings from being a lat,long pair and not necessarily that they are a lat,long pair. For example, "13" _could be_ a lat,long pair, because it doesn't have any disqualifying characters in it, though it is not a valid lat,long pair.